src/PostgREST/Error.hs

```diff
@@ -161,46 +192,52 @@ instance JSON.ToJSON PgError where
 
 instance JSON.ToJSON P.UsageError where
   toJSON (P.ConnectionError e) = JSON.object [
-    "code"    .= ("" :: Text),
+    "code"    .= ("PGCON" :: Text),
     "message" .= ("Database connection error. Retrying the connection." :: Text),
-    "details" .= (toSL $ fromMaybe "" e :: Text)]
+    "details" .= (toSL $ fromMaybe "" e :: Text),
+    "hint"    .= JSON.Null]
   toJSON (P.SessionError e) = JSON.toJSON e -- H.Error
 
 instance JSON.ToJSON H.QueryError where
   toJSON (H.QueryError _ _ e) = JSON.toJSON e
 
 instance JSON.ToJSON H.CommandError where
-  toJSON (H.ResultError (H.ServerError c m d h)) = case toS c of
-    'P':'T':_ -> JSON.object [
-        "details" .= (fmap toS d :: Maybe Text),
-        "hint"    .= (fmap toS h :: Maybe Text)]
-
-    _         -> JSON.object [
+  toJSON (H.ResultError (H.ServerError c m d h)) = JSON.object [
         "code"    .= (toS c      :: Text),
         "message" .= (toS m      :: Text),
         "details" .= (fmap toS d :: Maybe Text),
         "hint"    .= (fmap toS h :: Maybe Text)]
 
   toJSON (H.ResultError (H.UnexpectedResult m)) = JSON.object [
-    "message" .= (m :: Text)]
+    "code"    .= ("PGUNX" :: Text),
+    "message" .= (m :: Text),
+    "details" .= JSON.Null,
+    "hint"    .= JSON.Null]
   toJSON (H.ResultError (H.RowError i H.EndOfInput)) = JSON.object [
+    "code"    .= ("PGEOI" :: Text),
     "message" .= ("Row error: end of input" :: Text),
     "details" .= ("Attempt to parse more columns than there are in the result" :: Text),
     "hint"    .= (("Row number " <> show i) :: Text)]
   toJSON (H.ResultError (H.RowError i H.UnexpectedNull)) = JSON.object [
+    "code"    .= ("PGNUL" :: Text),
     "message" .= ("Row error: unexpected null" :: Text),
     "details" .= ("Attempt to parse a NULL as some value." :: Text),
     "hint"    .= (("Row number " <> show i) :: Text)]
   toJSON (H.ResultError (H.RowError i (H.ValueError d))) = JSON.object [
+    "code"    .= ("PGVAL" :: Text),
     "message" .= ("Row error: Wrong value parser used" :: Text),
     "details" .= d,
     "hint"    .= (("Row number " <> show i) :: Text)]
   toJSON (H.ResultError (H.UnexpectedAmountOfRows i)) = JSON.object [
+    "code"    .= ("PGUAR" :: Text),
     "message" .= ("Unexpected amount of rows" :: Text),
-    "details" .= i]
+    "details" .= i,
+    "hint"    .= JSON.Null]
   toJSON (H.ClientError d) = JSON.object [
+    "code"    .= ("PGCLI" :: Text),
```

Comment:
The only valuable code to differentiate would be this one I think. Because it's shown on a connection error which is not a PostgREST error per se.